_posts/2022-08-10-announcing-ballerina-2201.2.0-swan-lake-update-2.md

```diff
@@ -0,0 +1,122 @@
+- In terms of improvements, when a type is defined referring to another type, it will now be passed to the runtime as a `BTypeReferenceType` instance. A few existing runtime APIs have been modified to return the `BTypeReferenceType` instances.
+
+The following standard library and extended library modules now come with new features and improvements: 
+
+## `io`
+
+- Added support for data mapping between Ballerina Records and CSV in CSV read/write APIs
+
+## `constraint`
+
+- Introduced the `constraint` standard library package, which provides features to validate the values that have been assigned to Ballerina types
+
+## `http`
+
+- Added the initial support for HATEOAS
+- Added support for client resource methods in the HTTP client
+- Added proxy support for the HTTP2 client
+- Added constraint validation to HTTP payload binding
+- Made HTTP2 the default transport for the `http` module
+- Updated the default response status as `HTTP 201` for POST resources
+- Implemented the `immediateStop()` function for the HTTP listener
+
+## `websocket`
+
+- Added constraint validation to WebSocket payload binding
+
+## `graphql`
+
+- Added the support for deprecation of fields and enum values
+- Added the support for GraphQL interceptors
+
+## `serdes`
+
+- Introduced the `serdes` standard library module for serializing and deserializing Ballerina `anydata` subtypes
+- Proto3 is the initial serialization/deserialization format supported by this module
+
+## `os`
+
+- Introduced the `exec()` function to support OS command execution in Ballerina
+
+## `xmldata`
+
+- Introduced new APIs such as `fromXml` and `toXml` to perform conversions between XML and `map<anydata>`. The `toRecord` API is being deprecated by introducing this `fromXml` API
+- Introduced a new config named `rootTag` in the `JsonOptions` to configure the name of the XML root element tag
+
+ ## `random`
+
+- Updated the `createDecimal()` function to be cryptographically secure
+
+## `sql`
+
+- Added schema client abstraction to support metadata retrieval from SQL databases. The implementation for the connectors will be added soon
+
+## `grpc`
+
+- Introduced a message-level annotation for the proto descriptor instead of a centralized proto descriptor
+- Introduced a new service-level annotation (`grpc:Descriptor`) as a replacement for the existing (`grpc:ServiceDescriptor`) annotation. Both annotations are supported now to maintain backward compatibility. The `grpc:ServiceDescriptor` will be removed in the future. (Please update the service annotation if stub files are regenerated for the existing gRPC services)
+
+## `kafka`
+
+- Added constraint validation support for payload binding
+
+## `rabbitmq` 
+
+- Added constraint validation support for payload binding
+
+## `nats`
+
+- Added constraint validation support for payload binding
+
+Here are the major new features and improvements you can find in the developer tools:  
```

Comment:
Same as above. Shouldn't we have a sub title saying `Improvements to the developer tools`?

Reply:
Added.